# List of Exposures  at USDF in OGA


- work with Weakly_2024_17
- use jupyter kernel LSST


- author : Sylvie Dagoret-Campagne
- affiliation : IJCLab
- creation date : 2023/09/13
- last update : 2024/05/02



In [1]:
import sys
print(sys.executable)

/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-8.0.0/bin/python3


In [2]:
print(sys.path)

['/sdf/data/rubin/user/dagoret/AuxTelComm/notebooks_usdf/butlertools', '/home/d/dagoret/repos/repos_w_2024_16/Spectractor/lib/python', '/home/d/dagoret/repos/repos_w_2024_16/atmospec/python', '/home/d/dagoret/repos/repos_w_2024_16/analysis_tools/python', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linux64/summit_extras/g83f1fb8f92+e9fa2e932c/python', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linux64/summit_utils/gbd71fbde96+d024441011/python', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linux64/display_astrowidgets/g2bbee38e9b+d6b5cd48b5/python', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linux64/cell_coadds/g4da81e4fa4+4c30af3487/python', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linux64/gbdes/g73f919205b+ca4f911437/build/pydir', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linux64/drp_tasks/g3b9ebb4159+e4fcc55755/python', '/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-8.0.0/Linu

In [3]:
! eups list -s | grep LOCAL

analysis_tools        LOCAL:/home/d/dagoret/repos/repos_w_2024_16/analysis_tools 	setup
atmospec              LOCAL:/home/d/dagoret/repos/repos_w_2024_16/atmospec 	setup
eups                  LOCAL:/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-8.0.0/eups 	setup
spectractor           LOCAL:/home/d/dagoret/repos/repos_w_2024_16/Spectractor 	setup


In [4]:
! echo $IMAGE_DESCRIPTION
! eups list -s lsst_distrib

Weekly 2024_16
   g4213664e8e+c2911b1728 	current w_2024_16 setup


In [5]:
import os
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
%matplotlib inline
from matplotlib.colors import LogNorm

from mpl_toolkits.axes_grid1 import make_axes_locatable

import matplotlib.ticker                         # here's where the formatter is
from matplotlib.ticker import (MultipleLocator, FormatStrFormatter,
                               AutoMinorLocator)

from astropy.io import fits
from astropy.time import Time

In [6]:
from IPython.display import display, HTML

In [7]:
#import lsst.afw.display as afwDisplay
#afwDisplay.setDefaultBackend('matplotlib')

In [8]:
import lsst.daf.butler as dafButler

In [9]:
#repo = "/sdf/group/rubin/repo/main"
repo="/sdf/group/rubin/repo/oga/"
butler = dafButler.Butler(repo)
registry = butler.registry

In [10]:
collection='LATISS/raw/all'

## List of Exposures

In [11]:
df_exposure = pd.DataFrame(columns=['id', 'obs_id','day_obs', 'seq_num','time_start','time_end' ,'type', 'target','filter','zenith_angle','expos','ra','dec','skyangle','azimuth','zenith','science_program','jd','mjd'])

In [12]:
print(butler.registry.dimensions["exposure"].RecordClass.fields)

exposure: 
  instrument: str
  id: int
  day_obs: int
  group: str
  physical_filter: str
  obs_id: str
  exposure_time: float
  dark_time: float
  observation_type: str
  observation_reason: str
  seq_num: int
  seq_start: int
  seq_end: int
  target_name: str
  science_program: str
  tracking_ra: float
  tracking_dec: float
  sky_angle: float
  azimuth: float
  zenith_angle: float
  has_simulated: bool
  can_see_sky: bool
  timespan: lsst.daf.butler.Timespan


In [13]:
for count, info in enumerate(registry.queryDimensionRecords('exposure',where= "instrument='LATISS'")):
    
    
    try:
        df_exposure.loc[count] = [info.id, info.obs_id, info.day_obs, info.seq_num,pd.to_datetime(info.timespan.begin.to_string()),pd.to_datetime(info.timespan.end.to_string()) ,info.observation_type, info.target_name, info.physical_filter, info.zenith_angle, \
                             info.exposure_time,info.tracking_ra, info.tracking_dec, info.sky_angle,info.azimuth ,info.zenith_angle, info.science_program,
                             info.timespan.begin.jd,info.timespan.begin.mjd ]
    except:
        print(">>>   Unexpected error:", sys.exc_info()[0])
        info_timespan_begin_to_string = "2021-01-01 00:00:00.00"
        info_timespan_end_to_string = "2051-01-01 00:00:00.00"
        info_timespan_begin_jd = 0
        info_timespan_begin_mjd = 0
        df_exposure.loc[count] = [info.id, info.obs_id, info.day_obs, info.seq_num,
                                  pd.to_datetime(info_timespan_begin_to_string),
                                  pd.to_datetime(info_timespan_end_to_string) ,
                                  info.observation_type, info.target_name, 
                                  info.physical_filter, info.zenith_angle, \
                             info.exposure_time,info.tracking_ra, info.tracking_dec, info.sky_angle,info.azimuth ,info.zenith_angle, info.science_program,
                             info_timespan_begin_jd, info_timespan_begin_mjd  ]
 
    
    if count < 5:
        print("-----------------------------------------------------",count,"---------------------------------------------------------")
        print(info)
        print("\t id:                  ",info.id)
        print("\t day_obs:             ",info.day_obs)
        print("\t seq_num:             ",info.seq_num)
        print("\t type-of-observation: ",info.observation_type)
        print("\t target:              ",info.target_name)
        
        mjd = Time(info.timespan.begin.to_string()).mjd
        jd = Time(info.timespan.begin.to_string()).jd
        print(mjd,jd)
    

----------------------------------------------------- 0 ---------------------------------------------------------
exposure:
  instrument: 'LATISS'
  id: 2023011100002
  day_obs: 20230111
  group: '2023-01-12T04:10:45.662'
  physical_filter: 'BG40_65mm_1~empty'
  obs_id: 'AT_O_20230111_000002'
  exposure_time: 5.0
  dark_time: 5.23673
  observation_type: 'engtest'
  observation_reason: 'engtest'
  seq_num: 2
  seq_start: 2
  seq_end: 2
  target_name: 'azel_target'
  science_program: 'unknown'
  tracking_ra: 307.79861942556755
  tracking_dec: -20.253626608705332
  sky_angle: 100.23298943653465
  azimuth: 0.0013227050221757963
  zenith_angle: 9.999793858936755
  has_simulated: False
  can_see_sky: None
  timespan: Timespan(begin=astropy.time.Time('2023-01-12 04:10:47.090019', scale='tai', format='iso'), end=astropy.time.Time('2023-01-12 04:10:52.327000', scale='tai', format='iso'))
	 id:                   2023011100002
	 day_obs:              20230111
	 seq_num:              2
	 type-of-o

In [14]:
info

exposure.RecordClass(instrument='LATISS', id=2024031300005, day_obs=20240313, group='2024031300005', physical_filter='unknown~unknown', obs_id='AT_O_20240313_000005', exposure_time=0.0, dark_time=0.00866342, observation_type='bias', observation_reason='bias', seq_num=5, seq_start=5, seq_end=5, target_name='UNKNOWN', science_program='unknown', tracking_ra=None, tracking_dec=None, sky_angle=None, azimuth=None, zenith_angle=None, has_simulated=False, can_see_sky=False, timespan=Timespan(begin=astropy.time.Time('2024-03-13 12:13:42.007008', scale='tai', format='iso'), end=astropy.time.Time('2024-03-13 12:13:42.016000', scale='tai', format='iso')))

In [15]:
df_exposure

id                obs_id   day_obs  seq_num  \
0       2023011100002  AT_O_20230111_000002  20230111        2   
1       2023011200064  AT_O_20230112_000064  20230112       64   
2       2023011200063  AT_O_20230112_000063  20230112       63   
3       2023011200108  AT_O_20230112_000108  20230112      108   
4       2023011200110  AT_O_20230112_000110  20230112      110   
...               ...                   ...       ...      ...   
125509  2024031200457  AT_O_20240312_000457  20240312      457   
125510  2024031200459  AT_O_20240312_000459  20240312      459   
125511  2024031200461  AT_O_20240312_000461  20240312      461   
125512  2024031300003  AT_O_20240313_000003  20240313        3   
125513  2024031300005  AT_O_20240313_000005  20240313        5   

                       time_start                time_end     type  \
0      2023-01-12 04:10:47.090019 2023-01-12 04:10:52.327  engtest   
1      2023-01-13 04:46:30.894002 2023-01-13 04:46:31.920  engtest   
2      2023-01-13 04:45:59.695981 2023-01-13 04:46:00.927  engtest   
3      2023-01-13 05:04:31.147004 2023-01-13 05:04:32.180     flat   
4      2023-01-13 05:04:38.360000 2023-01-13 05:04:39.397     flat   
...                           ...                     ...      ...   
125509 2024-03-13 09:34:06.072990 2024-03-13 09:34:06.079     bias   
125510 2024-03-13 09:34:11.075980 2024-03-13 09:34:11.082     bias   
125511 2024-03-13 09:34:16.075993 2024-03-13 09:34:16.082     bias   
125512 2024-03-13 12:13:36.983016 2024-03-13 12:13:36.992     bias   
125513 2024-03-13 12:13:42.007008 2024-03-13 12:13:42.016     bias   

                    target             filter zenith_angle  expos          ra  \
0              azel_target  BG40_65mm_1~empty     9.999794    5.0  307.798619   
1       FlatField position  BG40_65mm_1~empty    51.125506    0.8   51.903967   
2       FlatField position  BG40_65mm_1~empty    51.125378    1.0   51.774217   
3       FlatField position  BG40_65mm_1~empty    51.126408    0.8        None   
4       FlatField position  BG40_65mm_1~empty    51.126411    0.8        None   
...                    ...                ...          ...    ...         ...   
125509             UNKNOWN    unknown~unknown         None    0.0        None   
125510             UNKNOWN    unknown~unknown         None    0.0        None   
125511             UNKNOWN    unknown~unknown         None    0.0        None   
125512             UNKNOWN    unknown~unknown         None    0.0        None   
125513             UNKNOWN    unknown~unknown         None    0.0        None   

              dec    skyangle     azimuth     zenith   science_program  \
0      -20.253627  100.232989    0.001323   9.999794           unknown   
1      -67.247896  286.766848  205.698886  51.125506  DayTime_Checkout   
2      -67.247837  286.766983  205.698885  51.125378  DayTime_Checkout   
3            None        None  205.698861  51.126408           unknown   
4            None        None  205.698861  51.126411           unknown   
...           ...         ...         ...        ...               ...   
125509       None        None        None       None           unknown   
125510       None        None        None       None           unknown   
125511       None        None        None       None           unknown   
125512       None        None        None       None           unknown   
125513       None        None        None       None           unknown   

                  jd           mjd  
0       2.459957e+06  59956.174156  
1       2.459958e+06  59957.198969  
2       2.459958e+06  59957.198608  
3       2.459958e+06  59957.211472  
4       2.459958e+06  59957.211555  
...              ...           ...  
125509  2.460383e+06  60382.398681  
125510  2.460383e+06  60382.398739  
125511  2.460383e+06  60382.398797  
125512  2.460383e+06  60382.509456  
125513  2.460383e+06  60382.509514  

[125514 rows x 19 columns]

In [16]:
df_exposure = df_exposure.astype({"id": int,'day_obs': int,'seq_num':int})

In [17]:
df_exposure

id                obs_id   day_obs  seq_num  \
0       2023011100002  AT_O_20230111_000002  20230111        2   
1       2023011200064  AT_O_20230112_000064  20230112       64   
2       2023011200063  AT_O_20230112_000063  20230112       63   
3       2023011200108  AT_O_20230112_000108  20230112      108   
4       2023011200110  AT_O_20230112_000110  20230112      110   
...               ...                   ...       ...      ...   
125509  2024031200457  AT_O_20240312_000457  20240312      457   
125510  2024031200459  AT_O_20240312_000459  20240312      459   
125511  2024031200461  AT_O_20240312_000461  20240312      461   
125512  2024031300003  AT_O_20240313_000003  20240313        3   
125513  2024031300005  AT_O_20240313_000005  20240313        5   

                       time_start                time_end     type  \
0      2023-01-12 04:10:47.090019 2023-01-12 04:10:52.327  engtest   
1      2023-01-13 04:46:30.894002 2023-01-13 04:46:31.920  engtest   
2      2023-01-13 04:45:59.695981 2023-01-13 04:46:00.927  engtest   
3      2023-01-13 05:04:31.147004 2023-01-13 05:04:32.180     flat   
4      2023-01-13 05:04:38.360000 2023-01-13 05:04:39.397     flat   
...                           ...                     ...      ...   
125509 2024-03-13 09:34:06.072990 2024-03-13 09:34:06.079     bias   
125510 2024-03-13 09:34:11.075980 2024-03-13 09:34:11.082     bias   
125511 2024-03-13 09:34:16.075993 2024-03-13 09:34:16.082     bias   
125512 2024-03-13 12:13:36.983016 2024-03-13 12:13:36.992     bias   
125513 2024-03-13 12:13:42.007008 2024-03-13 12:13:42.016     bias   

                    target             filter zenith_angle  expos          ra  \
0              azel_target  BG40_65mm_1~empty     9.999794    5.0  307.798619   
1       FlatField position  BG40_65mm_1~empty    51.125506    0.8   51.903967   
2       FlatField position  BG40_65mm_1~empty    51.125378    1.0   51.774217   
3       FlatField position  BG40_65mm_1~empty    51.126408    0.8        None   
4       FlatField position  BG40_65mm_1~empty    51.126411    0.8        None   
...                    ...                ...          ...    ...         ...   
125509             UNKNOWN    unknown~unknown         None    0.0        None   
125510             UNKNOWN    unknown~unknown         None    0.0        None   
125511             UNKNOWN    unknown~unknown         None    0.0        None   
125512             UNKNOWN    unknown~unknown         None    0.0        None   
125513             UNKNOWN    unknown~unknown         None    0.0        None   

              dec    skyangle     azimuth     zenith   science_program  \
0      -20.253627  100.232989    0.001323   9.999794           unknown   
1      -67.247896  286.766848  205.698886  51.125506  DayTime_Checkout   
2      -67.247837  286.766983  205.698885  51.125378  DayTime_Checkout   
3            None        None  205.698861  51.126408           unknown   
4            None        None  205.698861  51.126411           unknown   
...           ...         ...         ...        ...               ...   
125509       None        None        None       None           unknown   
125510       None        None        None       None           unknown   
125511       None        None        None       None           unknown   
125512       None        None        None       None           unknown   
125513       None        None        None       None           unknown   

                  jd           mjd  
0       2.459957e+06  59956.174156  
1       2.459958e+06  59957.198969  
2       2.459958e+06  59957.198608  
3       2.459958e+06  59957.211472  
4       2.459958e+06  59957.211555  
...              ...           ...  
125509  2.460383e+06  60382.398681  
125510  2.460383e+06  60382.398739  
125511  2.460383e+06  60382.398797  
125512  2.460383e+06  60382.509456  
125513  2.460383e+06  60382.509514  

[125514 rows x 19 columns]

## Selection of science

In [18]:
df_science = df_exposure[df_exposure.type == 'science']

In [19]:
df_science.sort_values(by="id",ascending=True,inplace=True)

/tmp/ipykernel_5825/3986782663.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_science.sort_values(by="id",ascending=True,inplace=True)


In [20]:
df_science.head(60)

id                obs_id   day_obs  seq_num  \
124235  2022091200104  AT_O_20220912_000104  20220912      104   
124234  2022091200105  AT_O_20220912_000105  20220912      105   
115615  2022091200106  AT_O_20220912_000106  20220912      106   
115616  2022091200107  AT_O_20220912_000107  20220912      107   
124237  2022091200111  AT_O_20220912_000111  20220912      111   
124236  2022091200112  AT_O_20220912_000112  20220912      112   
115618  2022091200113  AT_O_20220912_000113  20220912      113   
115617  2022091200114  AT_O_20220912_000114  20220912      114   
124238  2022091200119  AT_O_20220912_000119  20220912      119   
124239  2022091200120  AT_O_20220912_000120  20220912      120   
115619  2022091200121  AT_O_20220912_000121  20220912      121   
115628  2022091200122  AT_O_20220912_000122  20220912      122   
124246  2022091200126  AT_O_20220912_000126  20220912      126   
124247  2022091200127  AT_O_20220912_000127  20220912      127   
115629  2022091200128  AT_O_20220912_000128  20220912      128   
115620  2022091200129  AT_O_20220912_000129  20220912      129   
124248  2022091200133  AT_O_20220912_000133  20220912      133   
124240  2022091200134  AT_O_20220912_000134  20220912      134   
115621  2022091200135  AT_O_20220912_000135  20220912      135   
115630  2022091200136  AT_O_20220912_000136  20220912      136   
124249  2022091200140  AT_O_20220912_000140  20220912      140   
124250  2022091200141  AT_O_20220912_000141  20220912      141   
115623  2022091200142  AT_O_20220912_000142  20220912      142   
115622  2022091200143  AT_O_20220912_000143  20220912      143   
124251  2022091200147  AT_O_20220912_000147  20220912      147   
124241  2022091200148  AT_O_20220912_000148  20220912      148   
115624  2022091200149  AT_O_20220912_000149  20220912      149   
115625  2022091200150  AT_O_20220912_000150  20220912      150   
70247   2022091200151  AT_O_20220912_000151  20220912      151   
70248   2022091200152  AT_O_20220912_000152  20220912      152   
124252  2022091200156  AT_O_20220912_000156  20220912      156   
124242  2022091200157  AT_O_20220912_000157  20220912      157   
115631  2022091200158  AT_O_20220912_000158  20220912      158   
115626  2022091200159  AT_O_20220912_000159  20220912      159   
124243  2022091200163  AT_O_20220912_000163  20220912      163   
124244  2022091200164  AT_O_20220912_000164  20220912      164   
115632  2022091200165  AT_O_20220912_000165  20220912      165   
115627  2022091200166  AT_O_20220912_000166  20220912      166   
70249   2022091200167  AT_O_20220912_000167  20220912      167   
70250   2022091200168  AT_O_20220912_000168  20220912      168   
124245  2022091200172  AT_O_20220912_000172  20220912      172   
124253  2022091200173  AT_O_20220912_000173  20220912      173   
115634  2022091200174  AT_O_20220912_000174  20220912      174   
115633  2022091200175  AT_O_20220912_000175  20220912      175   
124254  2022091200179  AT_O_20220912_000179  20220912      179   
124261  2022091200180  AT_O_20220912_000180  20220912      180   
115635  2022091200181  AT_O_20220912_000181  20220912      181   
115636  2022091200182  AT_O_20220912_000182  20220912      182   
124262  2022091200186  AT_O_20220912_000186  20220912      186   
124255  2022091200187  AT_O_20220912_000187  20220912      187   
115642  2022091200188  AT_O_20220912_000188  20220912      188   
115637  2022091200189  AT_O_20220912_000189  20220912      189   
124263  2022091200194  AT_O_20220912_000194  20220912      194   
124256  2022091200195  AT_O_20220912_000195  20220912      195   
115643  2022091200196  AT_O_20220912_000196  20220912      196   
115644  2022091200197  AT_O_20220912_000197  20220912      197   
124257  2022091200201  AT_O_20220912_000201  20220912      201   
124264  2022091200202  AT_O_20220912_000202  20220912      202   
115638  2022091200203  AT_O_20220912_000203  20220912      203   
115639  2022091200204  AT_O_20220912_000204  20220912      

In [21]:
df_science.tail(60)

id                obs_id   day_obs  seq_num  \
40661   2024072900479  AT_O_20240729_000479  20240729      479   
72274   2024072900480  AT_O_20240729_000480  20240729      480   
76539   2024072900481  AT_O_20240729_000481  20240729      481   
6382    2024072900482  AT_O_20240729_000482  20240729      482   
40662   2024072900483  AT_O_20240729_000483  20240729      483   
72275   2024072900484  AT_O_20240729_000484  20240729      484   
76540   2024072900485  AT_O_20240729_000485  20240729      485   
6383    2024072900502  AT_O_20240729_000502  20240729      502   
40679   2024072900503  AT_O_20240729_000503  20240729      503   
72276   2024072900504  AT_O_20240729_000504  20240729      504   
76541   2024072900505  AT_O_20240729_000505  20240729      505   
6384    2024072900506  AT_O_20240729_000506  20240729      506   
40680   2024072900507  AT_O_20240729_000507  20240729      507   
72277   2024072900508  AT_O_20240729_000508  20240729      508   
76542   2024072900509  AT_O_20240729_000509  20240729      509   
111944  2024072900513  AT_O_20240729_000513  20240729      513   
111945  2024072900514  AT_O_20240729_000514  20240729      514   
80266   2024072900515  AT_O_20240729_000515  20240729      515   
80267   2024072900516  AT_O_20240729_000516  20240729      516   
80268   2024072900517  AT_O_20240729_000517  20240729      517   
111948  2024072900522  AT_O_20240729_000522  20240729      522   
111949  2024072900523  AT_O_20240729_000523  20240729      523   
80269   2024072900524  AT_O_20240729_000524  20240729      524   
80270   2024072900525  AT_O_20240729_000525  20240729      525   
80271   2024072900526  AT_O_20240729_000526  20240729      526   
111952  2024072900530  AT_O_20240729_000530  20240729      530   
111953  2024072900531  AT_O_20240729_000531  20240729      531   
80272   2024072900532  AT_O_20240729_000532  20240729      532   
80273   2024072900533  AT_O_20240729_000533  20240729      533   
80274   2024072900534  AT_O_20240729_000534  20240729      534   
111954  2024072900538  AT_O_20240729_000538  20240729      538   
111955  2024072900539  AT_O_20240729_000539  20240729      539   
80275   2024072900540  AT_O_20240729_000540  20240729      540   
80276   2024072900541  AT_O_20240729_000541  20240729      541   
80277   2024072900542  AT_O_20240729_000542  20240729      542   
111958  2024072900545  AT_O_20240729_000545  20240729      545   
111959  2024072900546  AT_O_20240729_000546  20240729      546   
80278   2024072900547  AT_O_20240729_000547  20240729      547   
80279   2024072900548  AT_O_20240729_000548  20240729      548   
80280   2024072900549  AT_O_20240729_000549  20240729      549   
111960  2024072900553  AT_O_20240729_000553  20240729      553   
111961  2024072900554  AT_O_20240729_000554  20240729      554   
80281   2024072900555  AT_O_20240729_000555  20240729      555   
80282   2024072900556  AT_O_20240729_000556  20240729      556   
80283   2024072900557  AT_O_20240729_000557  20240729      557   
111964  2024072900560  AT_O_20240729_000560  20240729      560   
111965  2024072900561  AT_O_20240729_000561  20240729      561   
80284   2024072900562  AT_O_20240729_000562  20240729      562   
80285   2024072900563  AT_O_20240729_000563  20240729      563   
80286   2024072900564  AT_O_20240729_000564  20240729      564   
111966  2024072900567  AT_O_20240729_000567  20240729      567   
111967  2024072900568  AT_O_20240729_000568  20240729      568   
80287   2024072900569  AT_O_20240729_000569  20240729      569   
80288   2024072900570  AT_O_20240729_000570  20240729      570   
80289   2024072900571  AT_O_20240729_000571  20240729      571   
111968  2024072900574  AT_O_20240729_000574  20240729      574   
111969  2024072900575  AT_O_20240729_000575  20240729      575   
80290   2024072900576  AT_O_20240729_000576  20240729      576   
80291   2024072900577  AT_O_20240729_000577  20240729      577   
80292   2024072900578  AT_O_20240729_000578  20240729      

In [22]:
df_science.day_obs.unique()

array([20220912, 20220913, 20220914, 20220927, 20220928, 20220929,
       20221011, 20221012, 20221013, 20221025, 20221026, 20221027,
       20221108, 20221109, 20221110, 20221122, 20221123, 20221124,
       20221207, 20221208, 20221209, 20221210, 20221212, 20230117,
       20230118, 20230119, 20230131, 20230201, 20230202, 20230214,
       20230215, 20230216, 20230228, 20230301, 20230302, 20230314,
       20230315, 20230316, 20230509, 20230510, 20230511, 20230523,
       20230524, 20230525, 20230704, 20230705, 20230706, 20230718,
       20230720, 20230801, 20230802, 20230803, 20230815, 20230816,
       20230817, 20230829, 20230912, 20230913, 20230914, 20230926,
       20230927, 20230928, 20231010, 20231011, 20231107, 20231108,
       20231113, 20231114, 20231115, 20231116, 20231120, 20231121,
       20231127, 20231128, 20231129, 20231130, 20231204, 20231205,
       20231206, 20231207, 20231211, 20231212, 20231214, 20231215,
       20231219, 20231220, 20231221, 20231222, 20240129, 20240

## Selection of date


  array([20220912, 20220913, 20220914, 20220927, 20220928, 20220929,
       20221011, 20221012, 20221013, 20221025, 20221026, 20221027,
       20221108, 20221109, 20221110, 20221122, 20221123, 20221124,
       20221207, 20221208, 20221209, 20221210, 20221212, 20230117,
       20230118, 20230119, 20230131, 20230201, 20230202, 20230214,
       20230215, 20230216, 20230228, 20230301, 20230302, 20230314,
       20230315, 20230316, 20230509, 20230510, 20230511, 20230523,
       20230524, 20230525, 20230704, 20230705, 20230706, 20230718,
       20230720, 20230801, 20230802, 20230803, 20230815, 20230816,
       20230817, 20230829, 20230912, 20230913, 20230914, 20230926,
       20230927, 20230928, 20231010, 20231011, 20231107, 20231108,
       20231113, 20231114, 20231115, 20231116, 20231120, 20231121,
       20231127, 20231128, 20231129, 20231130, 20231204, 20231205,
       20231206, 20231207, 20231211, 20231212, 20231214, 20231215,
       20231219, 20231220, 20231221, 20231222, 20240129, 20240130,
       20240131, 20240205, 20240206, 20240207, 20240208, 20240212,
       20240213, 20240214, 20240215, 20240219, 20240220, 20240222,
       20240223, 20240226, 20240227, 20240228, 20240229, 20240301,
       20240302, 20240303, 20240304, 20240305, 20240306, 20240307,
       20240308, 20240311, 20240312, 20240313, 20240314, 20240319,
       20240321, 20240325, 20240326, 20240327, 20240328, 20240329,
       20240401, 20240402, 20240403, 20240404, 20240405, 20240408,
       20240409, 20240410, 20240416, 20240417, 20240418, 20240419,
       20240422, 20240423, 20240424, 20240506, 20240527, 20240528,
       20240530, 20240624, 20240625, 20240626, 20240708, 20240709,
       20240710, 20240711, 20240715, 20240716, 20240717, 20240722,
       20240724, 20240729])

### Add those

20220504 - Not existing

In [162]:

# Januaryr 2024
#date_sel = 20240129
#date_sel = 20240130
#date_sel = 20240131
#date_sel = 20230928
date_sel = 20240717

## Query Dimension records on Exposures

In [163]:
where_clause = f"instrument=\'LATISS\' AND exposure.day_obs={date_sel} AND exposure.science_program=\'spec\'"

In [164]:
results_exposures = registry.queryDimensionRecords('exposure',datasets="raw",collections = 'LATISS/raw/all',where=where_clause)

## Select date

In [165]:
df_science_selected = df_science[df_science.day_obs == date_sel ]

In [166]:
len(df_science_selected)

230

# List of filters

In [167]:
list_of_filters = df_science_selected['filter'].unique()
list_of_filters

array(['empty~holo4_003', 'empty~blue300lpmm_qn1'], dtype=object)

In [168]:
selected_filters = []
for filt in list_of_filters:
    #flag_sel = (filt.find('holo4') != -1) or (filt.find('ronchi90lpmm') != -1) or (filt.find('ronchi170lpmm') != -1)
    flag_sel = (filt.find('holo4') != -1) 
    #flag_sel = (filt.find('ronchi170') != -1) 
    if flag_sel:
        selected_filters.append(filt) 
selected_filters = np.array(selected_filters)
selected_filters       

array(['empty~holo4_003'], dtype='<U15')

## Selection of filter

- But now we have a loop on filters

In [169]:
#idx_sel_filt = 0
#selected_filter = selected_filters[0]
#selected_filter

## Final selection of exposure

- loop on selected filters

In [170]:
pd.set_option('display.max_rows', None)

In [171]:
all_def_science_selected_final = []

for selected_filter in selected_filters:

    print(selected_filter)
    df_science_selected_final = df_science_selected[df_science_selected["filter"] ==  selected_filter]
    
    #print(df_science_selected_final)
    all_def_science_selected_final.append(df_science_selected_final.copy())
    
    df = df_science_selected_final
    df_sel = df[['day_obs', 'seq_num']]
    
    
    NN = len(df)
    
    if NN>0:
        # build the line
        
        df_sel['line'] = df_sel.apply(lambda row: str(row['day_obs'])  + " " + str(row['seq_num']),axis = 1)
    
    
         # prepare output
        filename_out = "visitdispersers_"+ str(date_sel) + "_filt_" + selected_filter.replace('~','-') + ".list"
        print(filename_out)
        
        df_out = df_sel[["line"]]
        df_out.to_csv(filename_out, index=False, header=None)

empty~holo4_003
visitdispersers_20240717_filt_empty-holo4_003.list


/tmp/ipykernel_5825/3368292521.py:20: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_sel['line'] = df_sel.apply(lambda row: str(row['day_obs'])  + " " + str(row['seq_num']),axis = 1)


In [172]:
N = len(all_def_science_selected_final)

In [173]:
if N>=1:
    n0=len(all_def_science_selected_final[0])
    display(all_def_science_selected_final[0])

id                obs_id   day_obs  seq_num  \
110787  2024071700271  AT_O_20240717_000271  20240717      271   
110788  2024071700272  AT_O_20240717_000272  20240717      272   
110791  2024071700278  AT_O_20240717_000278  20240717      278   
110792  2024071700279  AT_O_20240717_000279  20240717      279   
110793  2024071700285  AT_O_20240717_000285  20240717      285   
110794  2024071700286  AT_O_20240717_000286  20240717      286   
110795  2024071700292  AT_O_20240717_000292  20240717      292   
110796  2024071700293  AT_O_20240717_000293  20240717      293   
110797  2024071700299  AT_O_20240717_000299  20240717      299   
110798  2024071700300  AT_O_20240717_000300  20240717      300   
110801  2024071700306  AT_O_20240717_000306  20240717      306   
110784  2024071700307  AT_O_20240717_000307  20240717      307   
110802  2024071700313  AT_O_20240717_000313  20240717      313   
110803  2024071700314  AT_O_20240717_000314  20240717      314   
110821  2024071700320  AT_O_20240717_000320  20240717      320   
110822  2024071700321  AT_O_20240717_000321  20240717      321   
110832  2024071700327  AT_O_20240717_000327  20240717      327   
110836  2024071700328  AT_O_20240717_000328  20240717      328   
110846  2024071700334  AT_O_20240717_000334  20240717      334   
110850  2024071700335  AT_O_20240717_000335  20240717      335   
110860  2024071700341  AT_O_20240717_000341  20240717      341   
110864  2024071700342  AT_O_20240717_000342  20240717      342   
110876  2024071700348  AT_O_20240717_000348  20240717      348   
110877  2024071700349  AT_O_20240717_000349  20240717      349   
110884  2024071700355  AT_O_20240717_000355  20240717      355   
110885  2024071700356  AT_O_20240717_000356  20240717      356   
110890  2024071700379  AT_O_20240717_000379  20240717      379   
110891  2024071700380  AT_O_20240717_000380  20240717      380   
110894  2024071700386  AT_O_20240717_000386  20240717      386   
110895  2024071700387  AT_O_20240717_000387  20240717      387   
110896  2024071700393  AT_O_20240717_000393  20240717      393   
110897  2024071700394  AT_O_20240717_000394  20240717      394   
110898  2024071700400  AT_O_20240717_000400  20240717      400   
110899  2024071700401  AT_O_20240717_000401  20240717      401   
110900  2024071700407  AT_O_20240717_000407  20240717      407   
110901  2024071700408  AT_O_20240717_000408  20240717      408   
110902  2024071700414  AT_O_20240717_000414  20240717      414   
110903  2024071700415  AT_O_20240717_000415  20240717      415   
110904  2024071700422  AT_O_20240717_000422  20240717      422   
110905  2024071700423  AT_O_20240717_000423  20240717      423   
110906  2024071700429  AT_O_20240717_000429  20240717      429   
110907  2024071700430  AT_O_20240717_000430  20240717      430   
110908  2024071700436  AT_O_20240717_000436  20240717      436   
110909  2024071700437  AT_O_20240717_000437  20240717      437   
110910  2024071700443  AT_O_20240717_000443  20240717      443   
110911  2024071700444  AT_O_20240717_000444  20240717      444   
110912  2024071700450  AT_O_20240717_000450  20240717      450   
110913  2024071700451  AT_O_20240717_000451  20240717      451   
110914  2024071700457  AT_O_20240717_000457  20240717      457   
110915  2024071700458  AT_O_20240717_000458  20240717      458   
110916  2024071700464  AT_O_20240717_000464  20240717      464   
110917  2024071700465  AT_O_20240717_000465  20240717      465   
110918  2024071700471  AT_O_20240717_000471  20240717      471   
110919  2024071700472  AT_O_20240717_000472  20240717      472   
110920  2024071700478  AT_O_20240717_000478  20240717      478   
110921  2024071700479  AT_O_20240717_000479  20240717      479   
110924  2024071700485  AT_O_20240717_000485  20240717      485   
110925  2024071700486  AT_O_20240717_000486  20240717      486   
110926  2024071700511  AT_O_20240717_000511  20240717      511   
110927  2024071700512  AT_O_20240717_000512  20240717      

In [78]:
if N>=2:
    n1=len(all_def_science_selected_final[1])
    display(all_def_science_selected_final[1])

In [37]:
if N>=3:
    n2=len(all_def_science_selected_final[2])
    display(all_def_science_selected_final[2])

In [38]:
if N>=4:
    n3=len(all_def_science_selected_final[3])
    display(all_def_science_selected_final[3])

In [39]:
if N>=5:
    n4=len(all_def_science_selected_final[4])
    display(all_def_science_selected_final[4])

In [40]:
df_forsave = pd.concat(all_def_science_selected_final)

In [41]:
filename_out_csv = "holosummary_"+ str(date_sel) + ".csv"
filename_out_excel = "holosummary_"+ str(date_sel) + ".xlsx"

In [42]:
filename_out_csv

'holosummary_20240729.csv'

In [43]:
df_forsave.to_csv(filename_out_csv) 

In [ ]:
#! pip install openpyxl

In [44]:
df_forsave.to_excel(filename_out_excel) 